scanner.go

```diff
@@ -127,91 +125,87 @@ func (s *scanner) iterSingle(iter Scannable) (int, error) {
 	if err != nil {
 		return 0, err
 	}
-
-	setPtrs(structFields, ptrs, outVal)
+	fillInZeroedPtrs(ptrs)
 
 	s.rowsScanned++
 	return 1, nil
 }
 
-// structFields matches the SelectStatement field names selected to names of
```

Comment:
This struct fields method goes through the list of selected fields in a select statement and matches them up with the relevant fields in the struct.

This method has been inlined into `generatePtrs` instead and we can get rid of this method completely